File: biddo_web/src/core/repositories/last-seen-auctions.ts
import { Auction } from '../domain/auction'
import { RequestMaker, RequestType } from '../services/request-maker'

class LastSeenAuctionsRepository {
  private basePath = '/lastSeen'

  public async load(page = 0, perPage = 10): Promise<Auction[]> {
    try {
      const response = (await RequestMaker.makeRequest({
        path: `${this.basePath}/${page}/${perPage}`,
        method: RequestType.GET,
      })) as Record<string, unknown>[]
      return response.map((auction: Record<string, unknown>) => Auction.fromJSON(auction))
    } catch (error) {
      console.error('Error loading last seen auctions:', error)
      return []
    }
  }

  public async store(auctionId: string): Promise<void> {
    try {
      await RequestMaker.makeRequest({
        path: this.basePath,
        method: RequestType.POST,
        payload: JSON.stringify({ auctionId }),
        contentType: 'application/json',
      })
    } catch (error) {
      console.error('Error storing last seen auction:', error)
    }
  }
}

const LastSeenAuctionsRepositoryInstance = new LastSeenAuctionsRepository()
export { LastSeenAuctionsRepositoryInstance as LastSeenAuctionsRepository }

File: biddo_client/lib/core/repositories/last_seen_auctions.dart
import 'package:biddo/core/models/auction.dart';
import 'package:get/get.dart';

import 'base.dart';

class LastSeenAuctionsRepository {
  final dio = Get.find<Api>();

  Future<RxList<Rx<Auction>>> load([int page = 0, int perPage = 10]) async {
    try {
      var response = await dio.api.get('/lastSeen/$page/$perPage');
      return RxList<Rx<Auction>>.from(
        response.data.map(
          (auction) => Auction.fromJSON(auction).obs,
        ),
      );
    } catch (error, stacktrace) {
      print('Error loading last seen auctions: $error, $stacktrace');
      return [] as RxList<Rx<Auction>>;
    }
  }

  Future<void> store(String auctionId) async {
    try {
      await dio.api.post('/lastSeen', data: {
        'auctionId': auctionId,
      });
    } catch (error, stacktrace) {
      print('Error storing last seen auction: $error, $stacktrace');
    }
  }
}
